File: app/src/main/java/pe/ebim/canchaslima/ui/screens/ExplorarScreen.kt
package pe.ebim.canchaslima.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.google.android.gms.maps.model.BitmapDescriptorFactory
import com.google.android.gms.maps.model.CameraPosition
import com.google.maps.android.compose.GoogleMap
import com.google.maps.android.compose.MapProperties
import com.google.maps.android.compose.Marker
import com.google.maps.android.compose.MarkerState
import com.google.maps.android.compose.rememberCameraPositionState
import pe.ebim.canchaslima.data.Deporte
import pe.ebim.canchaslima.data.SampleData

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ExplorarScreen() {
    val camara = rememberCameraPositionState {
        position = CameraPosition.fromLatLngZoom(SampleData.centroPiloto, 12.5f)
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Explorar canchas") },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.primary,
                    titleContentColor = Color.White
                )
            )
        }
    ) { padding ->
        Box(Modifier.padding(padding).fillMaxSize()) {
            GoogleMap(
                modifier = Modifier.fillMaxSize(),
                cameraPositionState = camara,
                properties = MapProperties(isMyLocationEnabled = false)
            ) {
                SampleData.canchas.forEach { cancha ->
                    val hue = if (cancha.deporte == Deporte.PADEL)
                        BitmapDescriptorFactory.HUE_AZURE
                    else
                        BitmapDescriptorFactory.HUE_GREEN
                    Marker(
                        state = MarkerState(position = cancha.ubicacion),
                        title = "${cancha.nombre} · S/ ${cancha.precioHora}/h",
                        snippet = "${cancha.club} · ${cancha.deporte.etiqueta} · ${cancha.distrito.etiqueta}" +
                            if (cancha.clubFundador) " · Club Fundador" else "",
                        icon = BitmapDescriptorFactory.defaultMarker(hue)
                    )
                }
            }

            Leyenda(
                modifier = Modifier
                    .align(Alignment.BottomCenter)
                    .padding(16.dp)
            )
        }
    }
}

@Composable
private fun Leyenda(modifier: Modifier = Modifier) {
    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(12.dp),
        tonalElevation = 3.dp,
        shadowElevation = 6.dp,
        color = MaterialTheme.colorScheme.surface
    ) {
        Column(Modifier.padding(horizontal = 14.dp, vertical = 10.dp)) {
            Text(
                "Zona piloto: San Borja · Surco · La Molina",
                style = MaterialTheme.typography.labelMedium,
                fontWeight = FontWeight.SemiBold
            )
            Spacer(Modifier.size(6.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
                PuntoLeyenda(Color(0xFF2D6CDF), "Pádel")
                PuntoLeyenda(Color(0xFF0E7C4A), "Tenis")
            }
        }
    }
}

@Composable
private fun PuntoLeyenda(color: Color, texto: String) {
    Row(verticalAlignment = Alignment.CenterVertically) {
        Box(Modifier.size(12.dp).clip(RoundedCornerShape(50)).background(color))
        Spacer(Modifier.width(6.dp))
        Text(texto, style = MaterialTheme.typography.labelSmall)
    }
}

File: lib/screens/explorar_screen.dart
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

import '../data/sample_data.dart';
import '../models/models.dart';
import '../theme.dart';

class ExplorarScreen extends StatelessWidget {
  const ExplorarScreen({super.key});

  Set<Marker> _marcadores() {
    return SampleData.canchas.map((c) {
      final hue = c.deporte == Deporte.padel
          ? BitmapDescriptor.hueAzure
          : BitmapDescriptor.hueGreen;
      final fundador = c.clubFundador ? ' · Club Fundador' : '';
      return Marker(
        markerId: MarkerId(c.id),
        position: c.ubicacion,
        icon: BitmapDescriptor.defaultMarkerWithHue(hue),
        infoWindow: InfoWindow(
          title: '${c.nombre} · S/ ${c.precioHora}/h',
          snippet: '${c.club} · ${c.deporte.etiqueta} · ${c.distrito.etiqueta}$fundador',
        ),
      );
    }).toSet();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Explorar canchas')),
      body: Stack(
        children: [
          GoogleMap(
            initialCameraPosition: const CameraPosition(
              target: SampleData.centroPiloto,
              zoom: 12.5,
            ),
            markers: _marcadores(),
            myLocationButtonEnabled: false,
            zoomControlsEnabled: false,
          ),
          const Align(
            alignment: Alignment.bottomCenter,
            child: Padding(
              padding: EdgeInsets.all(16),
              child: _Leyenda(),
            ),
          ),
        ],
      ),
    );
  }
}

class _Leyenda extends StatelessWidget {
  const _Leyenda();

  @override
  Widget build(BuildContext context) {
    return Material(
      elevation: 4,
      borderRadius: BorderRadius.circular(12),
      color: Colors.white,
      child: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 10),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              'Zona piloto: San Borja · Surco · La Molina',
              style: TextStyle(fontWeight: FontWeight.w600, fontSize: 12),
            ),
            const SizedBox(height: 6),
            Row(
              mainAxisSize: MainAxisSize.min,
              children: const [
                _Punto(azulPadel, 'Pádel'),
                SizedBox(width: 16),
                _Punto(verdeCancha, 'Tenis'),
              ],
            ),
          ],
        ),
      ),
    );
  }
}

class _Punto extends StatelessWidget {
  final Color color;
  final String texto;
  const _Punto(this.color, this.texto);

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        Container(
          width: 12,
          height: 12,
          decoration: BoxDecoration(color: color, shape: BoxShape.circle),
        ),
        const SizedBox(width: 6),
        Text(texto, style: const TextStyle(fontSize: 11)),
      ],
    );
  }
}
